src/Execution/ExecutionStrategy.php

```diff
@@ -173,38 +179,47 @@ protected function resolveField(
         $fieldNodes,
         $path)
     {
+        $result = [];
         /** @var FieldNode $fieldNode */
-        $fieldNode = $fieldNodes[0];
-
-        $field = $parentType->getFields()[$fieldNode->getNameValue()];
-
-        $inputValues = $fieldNode->getArguments() ?? [];
-
-        $args = [];
-
-        foreach ($inputValues as $value) {
-            if ($value instanceof ArgumentNode) {
-                $args[] = $value->getValue()->getValue();
-            } elseif ($value instanceof InputValueDefinitionNode) {
-                $args[] = $value->getDefaultValue()->getValue();
+        foreach ($fieldNodes as $fieldNode) {
+            $field       = $parentType->getFields()[$fieldNode->getNameValue()];
+            $inputValues = $fieldNode->getArguments() ?? [];
+            $args        = [];
+
+            foreach ($inputValues as $value) {
+                if ($value instanceof ArgumentNode) {
+                    $args[] = $value->getValue()->getValue();
+                } elseif ($value instanceof InputValueDefinitionNode) {
+                    $args[] = $value->getDefaultValue()->getValue();
+                }
             }
-        }
 
-        $result = $field->resolve(...$args);
+            $subResult = $field->resolve(...$args);
+
+            if (!empty($fieldNode->getSelectionSet())) {
+                $fields = $this->collectFields(
+                    $parentType,
+                    $fieldNode->getSelectionSet(),
+                    new \ArrayObject(),
+                    new \ArrayObject()
+                );
+
+                $data = $this->executeFields(
+                    $parentType,
+                    $source,
+                    $path,
+                    $fields
+                );
+
+                $subResult = array_merge_recursive(
```

Comment:
This can be a costy operation. We should add a TODO to benchmark it.

Reply:
I'm going to refactor this very soon :D